packages/zwave-js/src/lib/commandclass/SceneControllerConfigurationCC.ts

```diff
@@ -0,0 +1,490 @@
+				throwUnsupportedProperty(this.ccId, property);
+		}
+	};
+
+	public async disable(groupId: number): Promise<void> {
+		this.assertSupportsCommand(
+			SceneControllerConfigurationCommand,
+			SceneControllerConfigurationCommand.Set,
+		);
+
+		return this.set(groupId, 0, new Duration(0, "seconds"));
+	}
+
+	public async set(
+		groupId: number,
+		sceneId: number,
+		dimmingDuration: Duration,
+	): Promise<void> {
+		this.assertSupportsCommand(
+			SceneControllerConfigurationCommand,
+			SceneControllerConfigurationCommand.Set,
+		);
+
+		const cc = new SceneControllerConfigurationCCSet(this.driver, {
+			nodeId: this.endpoint.nodeId,
+			endpoint: this.endpoint.index,
+			groupId,
+			sceneId,
+			dimmingDuration,
+		});
+
+		await this.driver.sendCommand(cc, this.commandOptions);
+	}
+
+	public async getLastActivated(): Promise<
+		| {
+				groupId: number;
+				sceneId: number;
+				dimmingDuration: Duration;
+		  }
+		| undefined
+	> {
+		this.assertSupportsCommand(
+			SceneControllerConfigurationCommand,
+			SceneControllerConfigurationCommand.Get,
+		);
+		return this.get(0);
+	}
+
+	public async get(
+		groupId: number,
+	): Promise<
+		| {
+				groupId: number;
+				sceneId: number;
+				dimmingDuration: Duration;
+		  }
+		| undefined
+	> {
+		this.assertSupportsCommand(
+			SceneControllerConfigurationCommand,
+			SceneControllerConfigurationCommand.Get,
+		);
+
+		const cc = new SceneControllerConfigurationCCGet(this.driver, {
+			nodeId: this.endpoint.nodeId,
+			endpoint: this.endpoint.index,
+			groupId,
+		});
+		const response = await this.driver.sendCommand<SceneControllerConfigurationCCReport>(
+			cc,
+			this.commandOptions,
+		);
+
+		// Return value includes "groupId", because if get(0) is called
+		// the returned report will include the actual groupId of the
+		// last activated groupId / sceneId
+		if (response)
+			return pick(response, ["groupId", "sceneId", "dimmingDuration"]);
+	}
+}
+
+@commandClass(CommandClasses["Scene Controller Configuration"])
+@implementedVersion(1)
+export class SceneControllerConfigurationCC extends CommandClass {
+	declare ccCommand: SceneControllerConfigurationCommand;
+
+	public determineRequiredCCInterviews(): readonly CommandClasses[] {
+		// AssociationCC is required and MUST be interviewed
+		// before SceneControllerConfigurationCC to supply groupCount
+		return [
+			...super.determineRequiredCCInterviews(),
+			CommandClasses.Association,
+		];
+	}
+
+	public async interview(complete: boolean = true): Promise<void> {
+		const node = this.getNode()!;
+		const endpoint = this.getEndpoint()!;
+		const api = endpoint.commandClasses[
+			"Scene Controller Configuration"
+		].withOptions({
+			priority: MessagePriority.NodeQuery,
+		});
+
+		this.driver.controllerLog.logNode(node.id, {
+			message: `${this.constructor.name}: doing a ${
+				complete ? "complete" : "partial"
+			} interview...`,
+			direction: "none",
+		});
+
+		const groupCount = this.getGroupCountCached();
```

Comment:
This should just use the `Association CC` instance
```suggestion
		const groupCount = endpoint
			.createCCInstanceUnsafe<AssociationCC>(
				CommandClasses.Association,
			)
			?.getGroupCountCached() ?? 0;
```

Reply:
Since `getGroupCountCached()` is called three times (`interview`, `SceneControllerConfigurationCCSet`, and `SceneControllerConfigurationCCGet`), I changed `getGroupCountCached` to a `protected` function that forwards the call to the `Association CC` instance.